resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java

```diff
@@ -796,11 +796,11 @@ private void validateServiceType(String serviceType, String requestServiceHistor
 		if(serviceType!=null) {
 			List<String> serviceTypes = List.of(serviceType.split(","));
 			for (String service : serviceTypes) {
-				if (!service.equalsIgnoreCase(ResidentTransactionType.DATA_SHARE_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.SERVICE_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.ID_MANAGEMENT_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.DATA_UPDATE_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.AUTHENTICATION_REQUEST.toString())
+				if (!service.equalsIgnoreCase(ServiceType.DATA_SHARE_REQUEST.name())
```

Comment:
No need to check for each and every enum like this, create a single static method that will return getServiceTypeForStrig() where you will return Optional<ServiceType>.

Reply:
changes done